docs/current_docs/api/internals.mdx

```diff
@@ -144,58 +144,152 @@ As this example demonstrates, Dagger object IDs hold the state of their correspo
 
 GraphQL query resolution is triggered only when a leaf value (scalar) is requested. Dagger uses this feature of GraphQL to evaluate workflows "lazily".
 
-An example will make this clearer. First, navigate to [Webhook.site](https://webhook.site), a free online tool that lets you receive and log incoming HTTP requests. Obtain and copy your unique webhook URL, as shown below:
+In practice, this means that if you create a Dagger object (like a container or a directory) but never access its state - either by returning it to the Dagger CLI, reading data from it, or using it in a different Dagger Function - Dagger automatically skips it as part of its optimization process, thinking it's not needed. This is analogous to running a build command but never using the output binary; since the output is never used, Dagger optimizes away the build step.
 
-![Webhook.site URL](/img/current_docs/api/webhook-1.png)
+There are some cases where this behavior causes unexpected results, such as when the command to be run has external effects. In this case, the optimization can be forcefully disabled with the `sync` field.
 
-Then, execute the following query, replacing the placeholder with your unique webhook URL:
+Here are two examples to make this clearer.
 
-```graphql
-query {
-  container {
-    from(address: "alpine:latest") {
-      withExec(args: ["apk", "add", "curl"]) {
-        withExec(args: ["curl", "YOUR-WEBHOOK-URL"]) {
+1. Navigate to [Webhook.site](https://webhook.site), a free online tool that lets you receive and log incoming HTTP requests. Obtain and copy your unique webhook URL, as shown below:
+
+    ![Webhook.site URL](/img/current_docs/api/webhook-1.png)
+
+    Then, execute the following query, replacing the placeholder with your unique webhook URL:
+
+    ```graphql
```

Comment:
Would it make sense to also have this as SDK examples? I know this is for internal documentation but maybe worth considering?

Reply:
Yes, I considered it already but I'd mark it as a lower priority item. Opened an internal issue for this.